streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -281,10 +280,27 @@ private void shutdown() {
         log.info("Stream thread shutdown complete [" + this.getName() + "]");
     }
 
+    /**
+     * Compute the latency based on the previous marked timestamp,
+     * and update the marked timestamp with the current system timestamp.
+     *
+     * @param now Previous marked timestamp
+     * @return latency
+     */
+    private long computeLatency(long now) {
+        long then = now;
+        now = time.milliseconds();
+
+        return Math.max(now - then, 0);
```

Comment:
The idea is to keep track of the last recorded timestamp as `now`, and then upon `computeLatency`, get the current timestamp and compute the difference, and then override `now` as the last recorded timestamp as the current value.